src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -679,11 +678,12 @@ void validateCmekKey(FailureCollector failureCollector, Map<String, String> argu
    */
   boolean shouldConnect() {
     return !containsMacro(BigQuerySinkConfig.NAME_DATASET) &&
-      !containsMacro(BigQuerySinkConfig.NAME_TABLE) &&
-      !containsMacro(NAME_SERVICE_ACCOUNT_TYPE) &&
-      !(containsMacro(NAME_SERVICE_ACCOUNT_FILE_PATH) || containsMacro(NAME_SERVICE_ACCOUNT_JSON)) &&
-      !containsMacro(BigQuerySinkConfig.NAME_PROJECT) &&
-      !containsMacro(BigQuerySinkConfig.DATASET_PROJECT_ID) &&
+      !containsMacro(BigQuerySinkConfig.NAME_TABLE) && connection != null && 
+      !connection.containsMacro(BigQueryConnectorConfig.NAME_SERVICE_ACCOUNT_TYPE) &&
```

Comment:
can these be refactored to `connection != null && connection.canConnect()`?

Reply:
done